### DISCUSSION TITLE
Quarkus Kafka - Maven archetype

### DISCUSSION THREAD

**Discussion by gdav4610** [quarkusio/quarkus#26688] [Community]
Hi community,

I have been develping a Quarkus based Maven archetype intended for developers to easily create Kafka applications (Kafka consumers and Kafka Streams), and adopt Apache Kafka and Red Hat AMQ Streams.
 
The archetype code is based on what I learned on the excellent Red Hat course:
AD482 - Developing Event-driven Applications with Apache Kafka and Red Hat AMQ Streams

What would be the best steps in order to contribute, document, and keep upgrading it?

I attach a screenshot of the folder structure:
![imagen](https://user-images.githubusercontent.com/33963190/178547241-940bcc48-7d38-432d-ba93-2aa2835a9f59.png)

Thank you.

**Comment by gdav4610** [comment]
Why an archetype vs a code start?

The reason is because a Maven archetype allows developers to easily generate their own personalized code base according to best practices recommended by Red Hat.

By using an archetype to generate an application codebase, a developer could save up to 30 hours of design-development for each new application. 
The idea is that best practices and standards are already coded within the archetype, which must be continously updated.
Some of the benefits at the time of code generation step are:

- Adhering to a proven project, folder and file structure for a good separation of concerns, 
- Usage of the latest dependencies available/recommended, 
- Following naming conventions, 
- Boostrapping the main usage of some components and their configurations (convention over configuration),
- Taking advantage of additional code generation tools (in this case, quarkus-apicurio-registry-avro Java code generator). 

All these things could be achieved by a developer in less than 1 hour by using a Maven archetype.

An archetype is a pragmatic solution for organisations that want to scale the adoption of a set of technologies (for example Quarkus, Red Hat AMQ, Kafka, OpenShift, etc) in a standarized way.
It could also help developers focus and master their own functional requirements with the help of easy to find  "//TODO: "  task tags, without the need of learning new frameworks and libraries from zero.

Finally, the code generation stage could also fit within a Jenkins/automation pipeline, allowing Red Hat to deliver a more automated experience for developers long before the CI/CD stage! This could be key for many developers!

**Reply by maxandersen** [reply]
@gdav4610 sounds like you have some specific layout in mind that is not what codestarts do? 

From your screenshot it does not look like that hence why I asked why it could not be a codestart so it would work in all the environments quarkus works (various IDE's and specifically for gradle and maven).

I would suggest you publish the archetype in a repo for now to gather interest.
